test-data/unit/check-columns.test

```diff
@@ -178,6 +178,79 @@ if int():
     def g(x): # E:5: Function is missing a type annotation
         pass
 
+[case testPrettyFunctionMissingTypeAnnotation]
+# flags: --disallow-untyped-defs --pretty
+def function_with_long_name():
+    pass
+[out]
+main:2:1: error: Function is missing a return type annotation
+    def function_with_long_name():
+    ^
+main:2:1: note: Use "-> None" if function does not return a value
+
+[case testColumnEndFunctionMissingTypeAnnotation]
+# flags: --disallow-untyped-defs --show-error-end
+from typing import Any, Optional
+if int():
+    def f():
+        pass
+
+    def f_no_return(x: int, foo: Optional[str]):
+        pass
+
+    def f_default(x: int, foo: Optional[str] = None):
+        pass
+
+    def f_default_untyped(x, foo = None):
+        pass
+
+    def f_args(x, *args):
+        pass
+
+    def f_kwargs(x, *args, **kwargs):
+        pass
+[builtins fixtures/tuple.pyi]
+[out]
+main:4:5:4:5: error: Function is missing a return type annotation
+main:4:5:4:5: note: Use "-> None" if function does not return a value
+main:7:5:7:47: error: Function is missing a return type annotation
+main:10:5:10:52: error: Function is missing a return type annotation
+main:13:5:13:40: error: Function is missing a type annotation
+main:16:5:16:24: error: Function is missing a type annotation
+main:19:5:19:36: error: Function is missing a type annotation
+
+[case testColumnEndFunctionMissingTypeAnnotationWithReturnType]
+# flags: --disallow-untyped-defs --show-error-end
+def f() -> None:
+    pass
+
+def f_partially_typed(x: int, foo) -> None:
+    pass
+
+def f_untyped(x, foo, *args, **kwargs) -> None:
+    pass
+[builtins fixtures/tuple.pyi]
+[out]
+main:5:1:5:43: error: Function is missing a type annotation for one or more arguments
+main:8:1:8:47: error: Function is missing a type annotation for one or more arguments
+
+[case testColumnEndMultiline]
+# flags: --disallow-untyped-defs --warn-no-return --show-error-end
+def f(
+    x: int,
+    y: int,
+):
+    pass
+
+def g(
+    x: int,
+    y: int,
+) -> int:
+    x = 1
+[out]
+main:2:1:4:11: error: Function is missing a return type annotation
```

Comment:
It's unfortunate that the closing `):` isn't included in the range. I imagine that's hard to fix though.

Reply:
I agree. I unfortunately don't know how this could be fixed since it is unavailable in the AST.

For reference: I did find [`ast.get_source_segment`](https://docs.python.org/3/library/ast.html#ast.get_source_segment). However, it also returns the source code for a function definition node including the body.

Reply:
I understand. I'm still struggling with this case (as well as the one above where if the function has no annotations at all, we emit a one-character range); I'm worried this PR may lead to more confusing output in some cases. However, on balance it's probably fine and we can apply this change, but consider reverting it if there's a lot of negative feedback.

We could also consider looking at the first start position in the body of the function, but then the whitespace between the colon and the first statement would be included in the range.